M2/Macaulay2/packages/UpdatePackage.m2

```diff
@@ -0,0 +1,122 @@
+
+try get "!curl -h 2>&1" else error "please install CURL on your system";
+
+newPackage(
+    "UpdatePackage",
+    Version => "1.0", 
+    Date => "February 2, 2021",
+    Authors => {{Name => "Giovanni Staglianò",Email => "giovannistagliano@gmail.com"}},
+    Headline => "install the latest development version of packages",
+    Keywords => {"System"}
+)
```

Comment:
Have you tested this package to see if it can update all the files from a package with auxiliary files?

Reply:
Yes, I did a lot of testing before proposing the package. However, in the current implementation it doesn't work for those few packages such that their auxiliary files directory contains another directory.

Reply:
It doesn't even work if auxiliary files are added or renamed. It is clear that it is a function that helps but can be improved.

Reply:
Another option would be to use git to checkout just the needed files.  Something like this:

```m2
git = null
updatePackage  = (pkg, remote, branch) -> (
    if git === null then git = findProgram("git", "git --help");
    dir := temporaryFileName();
    makeDirectory dir;
    runProgram(git,
	"clone --depth 1 --filter=blob:none --no-checkout https://github.com/" |
	remote | "/M2 -b " | branch, RunDirectory => dir);
    runProgram(git, "checkout " | branch | " -- M2/Macaulay2/packages/" | pkg |
	"*", RunDirectory => dir | "/M2");
    installPackage(pkg, FileName => dir | "/M2/M2/Macaulay2/packages/" | pkg |
	".m2");
)
```

Reply:
Yes, it looks nice and better than mine. What should I pass as "remote"?

Reply:
> Yes, it looks nice and better than mine. What should I pass as "remote"?

"Macaulay2" would be the default value -- we're cloning `https://github.com/<remote>/M2`.

Reply:
Thanks, it really works very well. Now I remove the package UpdatePackage from this PR.

Reply:
> Yes, I did a lot of testing before proposing the package. However, in the current implementation it doesn't work for those few packages such that their auxiliary files directory contains another directory.

Giovanni, why doesn't that work?  Can "curl" not handle it for some reason?

Reply:
This was due to my simplified implementation, Curl has nothing to do with it. Anyway I deleted that package since the method suggested by Doug is simpler and works better.